e2e/libs/jest-reporter.js

```diff
@@ -0,0 +1,80 @@
+const fs = require('fs-extra');
+const path = require('path');
+const { fileLogger } = require('./logger');
+const winston = require("winston");
+
+module.exports = class JestReporter {
+
+  constructor(globalConfig, options) {
+    if (globalConfig.verbose === true) {
+      throw Error("Verbose must be false or Console messages won't be available.")
+    }
+    this._globalConfig = globalConfig;
+    this._options = options;
+    this.logDir = this._options.outputdir || 'logs/jest';
+    this.summaryFile = this._options.filename || 'test-results-summary.json';
+  }
+
+  onTestStart(test) {
+    const time = new Date().toLocaleTimeString();
+    console.info(`Running ${path.parse(test.path).name} at ${time}`);
+  }
+
+  // @ts-ignore
+  onTestResult(testRunConfig, testResult, runResults) {
+    // Get test name.
+    const today = new Date();
+    const testName = path.parse(testResult.testFilePath).name;
+    const testLog = `${this.logDir}/${testName}-${today.getFullYear()+'-'+(today.getMonth()+1)+'-'+today.getDate()}.log`;
+
+    const transports = new winston.transports.File({
+        filename: testLog,
+        options: { flags: 'w' },
+        handleExceptions: true,
+      });
+    fileLogger.clear().add(transports);
```

Comment:
Can you explain the decision to use a single logger and then replace the output file - instead of creating a separate logger for each test case? Could you not just create a new one right here rather than mutating shared global state?

Reply:
Actually, I'm creating a new instance of winston logger here. Creating a separate log file for each test case. I didn't want to include `fileLogger` here is because it might be reusable for other purpose in future.

Reply:
On second thought: I think I know what you meant. It's a good idea.
Moving `fileLogger` code to here. creating a new log file for every test.

Reply:
Hmm, I realized I asked for this, but am I understanding correctly that now the network log / console logs will not be in the individual test case log files? If so - would probably just revert to the previous approach if you can't use individual loggers there.

It will be very useful to have a clear separation of network logs across test cases - right now it's hard to tell since you have to scan the logs to know how far back you can look for the problematic test case.

Reply:
Test network and console logs will be saved in individual test log file. they're not lost. It should be easier to read the test log when a failure has happened.

Reply:
True, I do seem them in the Circle run. I'm confused as to how they are getting there though, since this initial jest setup seems to reference the global logger: https://github.com/all-of-us/workbench/pull/4660/files#diff-9771f32e200b100d3bf9acedfc38dae6209ad815c5e214a31719188ddd392f1dR128

Reply:
I see. That `logger` object is a separate winston logger that only logs to console. It doesn't have a file transport.
However, console logs are automatically saved by jest and later they can be retrieved in `onTestResult` function when `verbose: false`.